src/CNLoginServer.cpp

```diff
@@ -133,16 +150,30 @@ void CNLoginServer::login(CNSocket* sock, CNPacketData* data) {
 
     // is the account banned
     if (findUser.BannedUntil > getTimestamp())
-        return loginFail(LoginError::LOGIN_ERROR, userLogin, sock);
-
-    /* 
-     * calling this here to timestamp login attempt,
-     * in order to make duplicate exit sanity check work
-     */
-    Database::updateSelected(findUser.AccountID, findUser.Selected);
+    {
+        // send a custom error message
+        INITSTRUCT(sP_FE2CL_GM_REP_PC_ANNOUNCE, msg);
+
+        // ceiling devision
+        int64_t remainingDays = (findUser.BannedUntil-getTimestamp()) / 86400 + ((findUser.BannedUntil - getTimestamp()) % 86400 != 0);
+
+        std::string text = "Your account has been banned by game master\nReason: ";
```

Comment:
```suggestion
        std::string text = "Your account has been banned. \nReason: ";
```

I'm not big on the term "game master". Feels silly/pompous. Let's either say moderator or nothing at all, since it's clear from context.